### PR TITLE
Fix version issues with Node and iojs

### PR DESCRIPTION
**Pull request by arv**
Newer ndoejs and iojs does not support passing a string to exec
where it expects an option. Older node does not allow passing
an option object. Therefore we now require 0.12.

0.12 shows a warning due to usage of deprecated API in mocha.
This updates mocha to a newer version and keeps chai in sync.

### CODE REVIEW

**Comment by arv** [comment]
@domenic PTAL

@johnjbarton FYI

**Comment by domenic** [comment]
There's really no reason to use 0.12 over iojs... 0.10 has some argument for it, since it has wide deployment, but 0.12 is kind of a DOA release. http://hueniverse.com/2015/03/02/the-node-version-dilemma/

Otherwise LGTM.

**Comment by arv** [comment]
You are right. Switching back to 0.10 since it is silly to limit the usage when it just this small thing.

**Comment by arv** [comment]
@johnjbarton PTAL

**Comment by johnjbarton** [comment]
LGTM I guess you tried the browser testRunner.html.

**Comment by arv** [comment]
testRunner.html is not working for me. Something about `global`. Working on a fix.

File: test/modular/Mocha6.js [hunk 1]
```diff
@@ -40,12 +40,13 @@ export class Mocha6 extends Mocha {
 
   importFiles() {
     var promiseImports = this.files.map((file) => {
-      file = './' + file.replace(/\\/g, '/');;
-      this.suite.emit('pre-require', global, file, this);
+      file = './' + file.replace(/\\/g, '/');
+      // TODO(arv): Why do we need to use global here?
+      this.suite.emit('pre-require', Reflect.global, file, this);
```
Comment (arv) [inline test/modular/Mocha6.js]: @johnjbarton What's the deal here? Is this required by mocha?

Comment (johnjbarton) [inline test/modular/Mocha6.js]: Mocha uses this event as a trigger to attach the definitions of suite/before/after to the argument.  That's how our older tests get a definition for suite().   The es6-based tests should always be using the imported suite().  But the feature tests are not converted. 

(Mocha does this to allow different "UI" versions to be mixed I guess).

What problem did you encounter. This was all working for me.

Comment (arv) [inline test/modular/Mocha6.js]: I just don't like us relying on the global object unless we have to.

The problem I encountered was that global was undefined (as expected).

Comment (johnjbarton) [inline test/modular/Mocha6.js]: Let me see I can send a PR to fix this.

Comment (arv) [inline test/modular/Mocha6.js]: OK. I'll use Reflect.global for now.

**Pull request merged**